contracts/adapters/Tribute.sol

```diff
@@ -0,0 +1,355 @@
+     * @param tokenAddr The address of the ERC-20 tokens provided as tribute by the proposer.
+     * @param tributeAmount The amount of tribute tokens.
+     */
+    function _submitTributeProposal(
+        DaoRegistry dao,
+        bytes32 proposalId,
+        address applicant,
+        address proposer,
+        address tokenToMint,
+        uint256 requestAmount,
+        address tokenAddr,
+        uint256 tributeAmount
+    ) internal {
+        dao.submitProposal(proposalId);
+        proposals[address(dao)][proposalId] = ProposalDetails(
+            proposalId,
+            applicant,
+            proposer,
+            tokenToMint,
+            requestAmount,
+            tokenAddr,
+            tributeAmount
+        );
+    }
+
+    /**
+     * @notice Refunds tribute tokens to the proposer.
+     * @param tokenAddr The address of the ERC-20 tokens provided as tribute by the proposer.
+     * @param proposer The proposer address (who will receive back the tribute tokens).
+     * @param amount The amount of tribute tokens to be refunded.
+     */
+    function _refundTribute(
+        address tokenAddr,
+        address proposer,
+        uint256 amount
+    ) internal {
+        IERC20 token = IERC20(tokenAddr);
+        token.safeTransfer(proposer, amount);
+    }
+
+    /**
+     * @notice Adds DAO internal tokens to applicant's balance and creates a new member entry (if applicant is not already a member).
+     * @dev Internal tokens to be minted to the applicant must be registered with the DAO Bank.
+     * @dev The applicant member cannot be jailed.
+     * @param dao The DAO address.
+     * @param applicant The applicant address (who will receive the DAO internal tokens and become a member).
+     * @param proposer The proposer address (who will be refunded the tribute tokens if the minting of internal tokens fails).
+     * @param tokenToMint The address of the DAO internal token to be minted to the applicant.
+     * @param requestAmount The amount requested of DAO internal tokens.
+     * @param token The address of the ERC-20 tokens provided as tribute by the proposer.
+     * @param tributeAmount The amount of tribute tokens to be refunded if the minting of internal tokens fails.
+     */
+    function _mintTokensToMember(
+        DaoRegistry dao,
+        address applicant,
+        address proposer,
+        address tokenToMint,
+        uint256 requestAmount,
+        address token,
+        uint256 tributeAmount
+    ) internal {
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        require(
+            bank.isInternalToken(tokenToMint),
+            "it can only mint internal tokens"
+        );
+        require(
+            !dao.getMemberFlag(applicant, DaoRegistry.MemberFlag.JAILED),
+            "cannot process jailed member"
+        );
+
+        dao.potentialNewMember(applicant);
+
+        try bank.addToBalance(applicant, tokenToMint, requestAmount) {
```

Comment:
I believe this was done to handle any overflow issue, is that correct? If so, what about adding a minor comment to help understand why this function might fail?

Reply:
Good idea. Yes, the primary is concern is overflow. I added a comment here as well as to `Onboarding.sol`.

Reply:
sounds good!